wgsl/index.bs

```diff
@@ -8178,6 +8146,55 @@ Here are some examples showing this analysis in action:
 </div>
 The same example would also be invalid for the same reason if `continue` was replaced by `break`.
 
+# Assertions # {#assertions}
+
+An <dfn noexport>assertion</dfn> is a check that ensures a boolean condition is satisfied.
+
+<pre class=include>
+path: syntax/global_assert.syntax.bs.include
+</pre>
+
+WGSL defines one kind of [=assertion=]: the const assertion. A const assertion verifies a bool-typed
+const-expression, evaluated at shader-creation time. If that expression evaluates to false, a
+shader-creation error occurs; otherwise, there is no runtime effect. A const assertion can appear at
+module scope or as a statement (see [[#const-assert-statement]]).
+
+<pre class=include>
+path: syntax/const_assert.syntax.bs.include
+</pre>
+
+[=Type rule precondition=]:
+The expression [=shader-creation error|must=] be of [=bool=] type.
+
+## Const Assertion Statement ## {#const-assert-statement}
+
+A const assertion statement is an [=assertion=] that
+produces a [=shader-creation error=] if the expression evaluates to `false`.
+The expression [=shader-creation error|must=] be a [=const-expression=].
+The statement can satisfy [=statically accessed|static access=] conditions in
+a shader, but otherwise has no effect on the compiled shader.
+This statement can be used within [=function scope|functions=].
```

Comment:
Good to call out it can be at either scope here I think.

```suggestion
A const assertion can appear at
[=module scope=] or as a [=function scope=] [=statement=].
```